step-capstone/src/components/Sidebars/PreferencePopover.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from "react";
+import Popover from "@material-ui/core/Popover";
+import Typography from "@material-ui/core/Typography";
+
+export default function PreferencePopover(props) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  useEffect(() => {
+    if (props.anchorEl !== anchorEl) {
+      setAnchorEl(props.anchorEl);
+    }
+  }, [props.anchorEl]);
+
+  useEffect(() => {
+    let coordinates = {lat: 0, lng: 0}
+    if (props.slots !== null) {
+      if (props.slots.prevSlot !== null && props.slots.nextSlot !== null) {
+        coordinates.lat = (props.slots.prevSlot.lat + props.slots.nextSlot.lat)/2;
+        coordinates.lgn = (props.slots.prevSlot.lgn + props.slots.nextSlot.lgn)/2
+      } else if (props.slots.prevSlot !== null) {
+        coordinates.lat = props.slots.prevSlot.lat;
+        coordinates.lgn = props.slots.prevSlot.lgn;
+      } else if (props.slots.nextSlot !== null ) {
+        coordinates.lat = props.slots.nextSlot.lat;
+        coordinates.lgn = props.slots.nextSlot.lgn;
+      } else {
+        //TODO: implement when merge code with Emmie's preference form and autocomplete
+        //coordinates = props.trip_coordinates;
+      }
```

Comment:
Nit: I know you haven't actually done this yet, but I think this else might not be quite right, since this should be the usual behavior, not the default behavior of using the surrounding coordinates.

Reply:
I am not so clear about this comment. Could you kindly rephrase it for me?

Reply:
Yeah sorry that wasn't super clear :D What I was trying to say is that maybe the TODO should occur in a different case. However, I realized that I was wrong anyway so all good.